### DISCUSSION TITLE
Extreme slow reaction - Each setup screen takes minutes to load

### DISCUSSION THREAD

**Discussion by Kuto587** [jomjol/AI-on-the-edge-device#1124] [Q&A] [answered]
Hi,
I installed the Digitizer on an ESP32 and setup the SD-Card. Then I accessed the Setup-page via my browser. My problem is that the loading of each setup html-page takes a very long time. Several minutes for each of the setup screens. I setup the reference picture, set the image references and then set the digits for the numbers, but because of the long loading times for each picture which is transmitted from the ESP, this takes forever. Also for the analog readings this does not seem to work at all (I waited at least 10 minutes). Judging from the youtube online tutorials, this should be much quicker right? This is not the normal behavior, right?

I checked the SD-Card (SanDisk Ultra 8Gb), the PSRAM module (AP-Memory 6404L-3SOR) but that all seems fine. 
Could this be a problem with the power supply? Or perhaps is the Wifi network not right (I use a Vodafone router, which I suspect to cause some strange behaviour sometimes...)? Anyone any ideas?

Best,
Phil

**Comment by caco3** [comment, marked as answer]
Yes, the Web Interface usually responds impressively quickly. If is slow, then usually its due to a weak Wifi connection.
Looking on the main page, on the bottom right, you can see the RSSI value. That represents the wifi connection quality. What link do you see?

Could you try to position the ESP close to your accesspoint for a comparison?

It might also be required to replace the internal antenna with an external one.

**Reply by Kuto587** [reply]
Wow thanks for the super-quick reply!

I am not certain what do you mean by main-page? Due to the slow response, I never got past the setup...

**Reply by Kuto587** [reply]
In other words, is the RSSI-Value also present on the Setup-pages?

**Reply by caco3** [reply]
Ah, no sry, I didn't read it right.
That page only is visible after the setup. :(

Not sure if it already works during setup, but give it a try: `http://<IP>/rssi.html`
I see `RSSI: -80dBm` which is already low but still ok. Usually it is stated that it must be better (means closer to positive) than `-85 dBm`.

**Reply by Kuto587** [reply]
Okay, I placed it near my router (I failed to mention that before, I had used it near a repeater). The response is much quicker. I think the thing to blame is indeed my router... That creates new problems during setup... anyways, I think the take-away message here is that slow connections can arise due to uncompatible routers (as I said, the Vodafone routers are shit...)

**Reply by Kuto587** [reply]
Or rather, my setup of having a shitty router in combination with a repeater...

**Reply by caco3** [reply]
ok, great you found the issue!

Once you have it setup, could you tell us the RSSI? I would like to get a feeling for it.

Also, in case you are not yet aware of it, the ESP32-CAM supports an external antenna. It requires some soldering skills but can improve the connection quality. See https://randomnerdtutorials.com/esp32-cam-connect-external-antenna/

**Comment by pos-ei-don** [comment]
In my instaation, slow sites are not caused by bad rssi, some Browsers are. Using a different Browser like mobile Browsers work much much better. I didnt have time to debug it fully.

**Reply by caco3** [reply]
> In my instaation, slow sites are not caused by bad rssi, some Browsers are. Using a different Browser like mobile Browsers work much much better. I didnt have time to debug it fully.

ok, thats interesting. Can you mention the good/bad browsers? We know that the OTA also is picky on them.

**Comment by pos-ei-don** [comment]
Others saw this as well.
Chrome on Windows is slow, Chrome on Android is ok.
So Times even Firefox on Windows is slow. Repeatable and on different clients.
